### PR TITLE
Introduce `CSVConfig` structure

### PR DESCRIPTION
**Pull request by vs9h**
This pull request is a logical continuation of the pull request #303.

Currently, tables config that are passed to create input tables for algorithms that run on a set of tables (IND algorithms) are subject to some configuration restrictions: tables must have one separator and either all must have a header or none must have a header. This pull request adds the support to use different configurations for different tables. 

The use of new structure has also been added for algorithms that process single file. We managed to avoid code duplication associated with passing table configuration parameters.

I also made small fixes to the code of various tests along the way.

### CODE REVIEW

File: src/core/algorithms/algo_factory.cpp [hunk 1]
```diff
@@ -47,24 +47,19 @@ void LoadAlgorithm(Algorithm& algorithm, StdParamsMap const& options) {
         using namespace config::names;
         namespace fs = std::filesystem;
         if (option_name == kTable && options.find(std::string{kTable}) == options.end()) {
-            config::InputTable parser =
-                    std::make_shared<CSVParser>(GetOptionValue<fs::path>(options, kCsvPath),
-                                                GetOptionValue<char>(options, kSeparator),
-                                                GetOptionValue<bool>(options, kHasHeader));
-            return boost::any{parser};
+            config::InputTable table =
+                    std::make_shared<CSVParser>(GetOptionValue<CSVConfig>(options, kCsvConfig));
```
Comment (vs9h) [inline src/core/algorithms/algo_factory.cpp:51]:
```suggestion
            config::InputTable table = GetOptionValue<CSVConfig>(options, kCsvConfig).MakeInputTable();
```

File: src/core/algorithms/algo_factory.cpp [hunk 2]
```diff
@@ -47,24 +47,19 @@ void LoadAlgorithm(Algorithm& algorithm, StdParamsMap const& options) {
...
+                throw config::ConfigurationError("Expected collection of csv configs");
             }
 
             config::InputTables tables;
-            tables.reserve(paths.size());
-            for (auto const& path : paths) {
-                tables.push_back(std::make_shared<CSVParser>(path, separator, has_header));
+            tables.reserve(configs.size());
+            for (auto const& config : configs) {
+                tables.push_back(std::make_shared<CSVParser>(config));
```
Comment (vs9h) [inline src/core/algorithms/algo_factory.cpp:62]:
```suggestion
                tables.push_back(config.MakeInputTable());
```

File: src/tests/test_cfd_relation_data.cpp [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
 
-static fs::path ConstructPath(std::string_view dataset) {
-    return test_data_dir / "cfd_data" / dataset;
-}
-
 TEST(TestCFDRelationData, TennisDataSet) {
-    std::string_view tennis_path = "tennis.csv";
-    CSVParser parser{ConstructPath(tennis_path), ',', true};
+    auto input_table = MakeInputTable(ktennis);
+    ;
```
Comment (vs9h) [inline src/tests/test_cfd_relation_data.cpp:15]: remove this line

File: src/tests/test_metric_verifier.cpp [hunk 4]
```diff
@@ -132,87 +124,90 @@ TEST_P(TestHighlights, DefaultTest) {  // Assumes that highlights are sorted by
...
-                MetricVerifyingParams("euclidean", 20500, {0}, {4}, "TestMetric.csv"),
-                MetricVerifyingParams("euclidean", 1059, {1}, {4}, "TestMetric.csv"),
-                MetricVerifyingParams("euclidean", 1, {1, 0}, {4}, "TestMetric.csv"),
-                MetricVerifyingParams("euclidean", 4.5724231, {0}, {2}, "TestMetric.csv"),
-                MetricVerifyingParams("euclidean", 7.53, {0}, {3}, "TestMetric.csv"),
-                MetricVerifyingParams("levenshtein", 2, {0}, {5}, "TestMetric.csv"),
-                MetricVerifyingParams("levenshtein", 3, {1}, {5}, "TestMetric.csv"),
-                MetricVerifyingParams("levenshtein", 4, {0}, {6}, "TestMetric.csv"),
-                MetricVerifyingParams("levenshtein", 10, {0}, {6}, "TestMetric.csv", "brute", true,
+                MetricVerifyingParams(kTestLong, "euclidean", 2, {0, 1}, {2}),
```
[the remark this answers is not in the document]
Comment (vs9h) [reply src/tests/test_metric_verifier.cpp]: I think, that this is out of scope, but I fixed it (the only thing is that I simply passed the enums explicitly)

**Review by Mstrutov — APPROVED** [review]
LGTM

**Pull request merged**